### ISSUE TITLE
tutorial/demo for 2D slice analysis based on 3D inputs

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/tutorials#130]
**Is your feature request related to a problem? Please describe.**
would be great to demonstrate the available options for both training and inference

there were quite a few requests, see also:
- https://github.com/Project-MONAI/MONAI/discussions/1641
- https://github.com/Project-MONAI/MONAI/issues/719
- https://github.com/Project-MONAI/MONAI/issues/995
- https://github.com/Project-MONAI/MONAI/issues/808

**Comment by rijobro** [comment]
Feel free to use this if it's of use to you.

For the [inverse transformation / test time augmentation notebook](https://github.com/Project-MONAI/tutorials/pull/129), I created a 2D dataset from the 3D brain tumour dataset. I use the slice in the 3D that had the largest segmented volume.

I then saved to file, which obviously isn't necessary, but I found it to be faster. Output is then a list of dictionaries with filenames to the 2D slices: `{"image": im_2d_fname, "label": label_2d_fname}`. Filenames are preserved from the original images' metadata.

```python
class SliceWithMaxNumLabelsd(MapTransform):
    def __init__(self, keys, label_key):
        self.keys = keys
        self.label_key = label_key

    def __call__(self, data):
        d = dict(data)
        im = d[self.label_key]
        q = np.sum((im > 0).reshape(-1, im.shape[-1]), axis=0)
        _slice = np.where(q == np.max(q))[0][0]
        for key in self.keys:
            d[key] = d[key][..., _slice]
        return d


class SaveSliced(MapTransform):
    def __init__(self, keys, path):
        self.keys = keys
        self.path = path

    def __call__(self, data):
        d = {}
        for key in self.keys:
            fname = os.path.basename(
                data[key + "_meta_dict"]["filename_or_obj"])
            path = os.path.join(self.path, key, fname)
            nib.save(nib.Nifti1Image(data[key], np.eye(4)), path)
            d[key] = path
        return d


keys = ["image", "label"]
data_dir = os.path.join(root_dir, task + "_single_slice")
for key in keys:
    os.makedirs(os.path.join(data_dir, key), exist_ok=True)
transform_2d_slice = Compose([
    LoadImaged(keys),
    AsChannelFirstd("image"),
    AddChanneld("label"),
    SliceWithMaxNumLabelsd(keys, "label"),
    SaveSliced(keys, data_dir),
])
ds_2d = Dataset(data_dicts, transform_2d_slice)
dl_2d = DataLoader(ds_2d, batch_size=1, num_workers=10)
data_dicts_single_slice = list(tqdm(dl_2d))
```

**Comment by rijobro** [comment]
`SliceWithMaxNumLabelsd` could be generalised and added to MONAI if you'd like. By generalised I mean the user could choose along which spatial axis they want to slice.

**Comment by wyli** [comment]
thanks, I think recently we have a `SaveImage` transform implemented, might be useful here as well

**Comment by rijobro** [comment]
Certainly could, I'll look into updating my PR!

I just noticed that when you refer to a Discussion (e.g., your first link), that discussion doesn't get a link pointing to the issue/PR discussing it. This is a shame, since the people who posted those questions won't necessarily know we're discussing it here.

**Comment by hasansari** [comment]
Hello, I am just wondering if there is a new transform implemented in Monai to get 2D samples from 3D volumes. I use RandWeightedCropd during the training to feed random slices into my 2D network. My question is if there is any function to sequentially get all test all of the slices of a 3D volume during the inference stage so I can save the output as a 3D volume?

Thanks in advance!
